spec/factories/submission_entry.rb

```diff
@@ -12,5 +21,13 @@
     factory :errored_submission_entry do
       aasm_state :errored
     end
+
+    factory :validated_invoice_submission_entry, parent: :validated_submission_entry do
+      invoice
+    end
+
+    factory :validated_order_submission_entry, parent: :validated_submission_entry do
+      order
+    end
```

Comment:
May be beyond the scope of this PR, but if we're going to use traits could we combine them as a list of traits at the call site rather than in factory names? For example, if we were to make `:validated` and `:errored` traits as well as `:invoice` and `:order` as done here we could be saying `FactoryBot.create :submission_entry, :validated, :invoice` rather than `FactoryBot.create :validated_invoice_submission_entry`. It's not a huge issue unless our combinations explode :)